### PR TITLE
Added StateLayout docs

### PR DESCRIPTION
**Pull request by sthewissen**
## What changes to the docs does this PR provide?
- Adds a page for `StateLayout`

## PR Checklist

Please check if your PR fulfills the following requirements:

- [X] For new pages, used the [provided template](https://github.com/MicrosoftDocs/xamarin-communitytoolkit/blob/master/docs/.template.md).
- [X] For new features, added an entry in the [Table of Contents](https://github.com/MicrosoftDocs/xamarin-communitytoolkit/blob/master/docs/toc.yml).
- [X] Ran against a spell and grammar checker.
- [X] Contains **NO** breaking changes.

### CODE REVIEW

**Comment by cmcclister** [comment]
Docs Build status updates of commit _[b85a61e](https://github.com/sthewissen/xamarin-communitytoolkit/commits/b85a61e0952729940254f7724beeb20bd0695f11)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/TOC.yml](https://github.com/sthewissen/xamarin-communitytoolkit/blob/statelayout/docs/TOC.yml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/xamarin/communitytoolkit/views/statelayout?branch=pr-en-us-28) |
[docs/views/statelayout.md](https://github.com/sthewissen/xamarin-communitytoolkit/blob/statelayout/docs/views/statelayout.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/xamarin/communitytoolkit/views/statelayout?branch=pr-en-us-28) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C10%5C31%5C52490272-7c0a-c2c6-3ec7-0fb104146cb5%5CPullRequest%5C202010310915129115-28%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=320NpG%2FSvlkAl6rciG%2Fib8l%2FEf8uI05x%2FZ4HHvNExYI%3D&st=2020-10-31T09%3A11%3A10Z&se=2020-12-01T09%3A16%3A10Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/52490272-7c0a-c2c6-3ec7-0fb104146cb5?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching in the <a href="https://review.docs.microsoft.com/en-us/search/?search=&amp;category=All&amp;scope=help-docs&amp;category=All&amp;branch=master">Docs contributor and Admin Guide</a></li><li>See the <a href="https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master">frequently asked questions</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

File: docs/views/statelayout.md [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+## Sample
+
+[StateLayout sample page Source](https://github.com/xamarin/XamarinCommunityToolkit/blob/main/XamarinCommunityToolkitSample/Pages/Views/StateLayoutPage.xaml)
+
+You can see this in action in the [Xamarin Community Toolkit Sample App](https://github.com/xamarin/XamarinCommunityToolkit).
+
+## API
+
+* [StateLayout source code](https://github.com/xamarin/XamarinCommunityToolkit/blob/main/XamarinCommunityToolkit/Views/StateLayout.shared.cs)
```
Comment (Kapusch) [inline docs/views/statelayout.md]:
```suggestion
* [StateLayout source code](https://github.com/xamarin/XamarinCommunityToolkit/blob/main/XamarinCommunityToolkit/Views/StateLayout/StateLayout.shared.cs)
```

**Comment by cmcclister** [comment]
Docs Build status updates of commit _[4cee038](https://github.com/sthewissen/xamarin-communitytoolkit/commits/4cee038bcd9047d875d8a44b76be97ce4d3253a6)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/TOC.yml](https://github.com/sthewissen/xamarin-communitytoolkit/blob/statelayout/docs/TOC.yml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/xamarin/communitytoolkit/views/statelayout?branch=pr-en-us-28) |
[docs/views/statelayout.md](https://github.com/sthewissen/xamarin-communitytoolkit/blob/statelayout/docs/views/statelayout.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/xamarin/communitytoolkit/views/statelayout?branch=pr-en-us-28) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C11%5C27%5C52490272-7c0a-c2c6-3ec7-0fb104146cb5%5CPullRequest%5C202011271226552122-28%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=JWsGZLqRYD%2Fh6DTlwVsvAZ3HUWCaWwaIrIT8ttJ85d0%3D&st=2020-11-27T12%3A22%3A45Z&se=2020-12-28T12%3A27%3A45Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/52490272-7c0a-c2c6-3ec7-0fb104146cb5?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching in the <a href="https://review.docs.microsoft.com/en-us/search/?search=&amp;category=All&amp;scope=help-docs&amp;category=All&amp;branch=master">Docs contributor and Admin Guide</a></li><li>See the <a href="https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master">frequently asked questions</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

**Comment by davidbritch** [comment]
Docs Build status updates of commit _[3783537](https://github.com/sthewissen/xamarin-communitytoolkit/commits/37835379cfcb9f739d2a45d47deb3f40e78610eb)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/TOC.yml](https://github.com/sthewissen/xamarin-communitytoolkit/blob/statelayout/docs/TOC.yml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/xamarin/communitytoolkit/views/statelayout?branch=pr-en-us-28) |
[docs/views/statelayout.md](https://github.com/sthewissen/xamarin-communitytoolkit/blob/statelayout/docs/views/statelayout.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/xamarin/communitytoolkit/views/statelayout?branch=pr-en-us-28) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C12%5C7%5C52490272-7c0a-c2c6-3ec7-0fb104146cb5%5CPullRequest%5C202012071226302669-28%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=pEuu2L5svQ4cZStXu4%2BRf8rCJAryuYIeSWxaTQ1UcNc%3D&st=2020-12-07T12%3A22%3A06Z&se=2021-01-07T12%3A27%3A06Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/52490272-7c0a-c2c6-3ec7-0fb104146cb5?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching in the <a href="https://review.docs.microsoft.com/en-us/search/?search=&amp;category=All&amp;scope=help-docs&amp;category=All&amp;branch=master">Docs contributor and Admin Guide</a></li><li>See the <a href="https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master">frequently asked questions</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

**Review by davidbritch — APPROVED** [review]
Thanks @sthewissen

**Comment by davidbritch** [comment]
Docs Build status updates of commit _[ec1cbeb](https://github.com/sthewissen/xamarin-communitytoolkit/commits/ec1cbeb9602bf126b2b7c73b4e993b16499fbcca)_: 

### :white_check_mark: Validation status: passed


File | Status | Preview URL | Details
---- | ------ | ----------- | -------
[docs/TOC.yml](https://github.com/sthewissen/xamarin-communitytoolkit/blob/statelayout/docs/TOC.yml) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/xamarin/communitytoolkit/views/statelayout?branch=pr-en-us-28) |
[docs/views/statelayout.md](https://github.com/sthewissen/xamarin-communitytoolkit/blob/statelayout/docs/views/statelayout.md) | :white_check_mark:Succeeded | [View](https://review.docs.microsoft.com/en-us/xamarin/communitytoolkit/views/statelayout?branch=pr-en-us-28) |

For more details, please refer to the [build report](https://opbuildstorageprod.blob.core.windows.net/report/2020%5C12%5C7%5C52490272-7c0a-c2c6-3ec7-0fb104146cb5%5CPullRequest%5C202012071234437012-28%5Cworkflow_report.html?sv=2016-05-31&sr=b&sig=CLXjv6LT9DBtBIjNV53e4p9zJpXdCU2E7Krl%2BUF1Pk4%3D&st=2020-12-07T12%3A30%3A56Z&se=2021-01-07T12%3A35%3A56Z&sp=r).

**Note:** Broken links written as relative paths are included in the above build report. For broken links written as absolute paths or external URLs, see the [broken link report](https://ops.microsoft.com/#/repos/52490272-7c0a-c2c6-3ec7-0fb104146cb5?tabName=brokenlinks).

For any questions, please:<ul><li>Try searching in the <a href="https://review.docs.microsoft.com/en-us/search/?search=&amp;category=All&amp;scope=help-docs&amp;category=All&amp;branch=master">Docs contributor and Admin Guide</a></li><li>See the <a href="https://review.docs.microsoft.com/en-us/help/onboard/faq?branch=master">frequently asked questions</a></li><li>Post your question in the <a href="https://teams.microsoft.com/l/channel/19%3a7ecffca1166a4a3986fed528cf0870ee%40thread.skype/General?groupId=de9ddba4-2574-4830-87ed-41668c07a1ca&amp;tenantId=72f98bf-86f1-41af-91ab-2d7cd011db47">Docs support channel</a></li></ul>

**Pull request merged**